### PR TITLE
qgsprojectelevationsettingswidget: Check elevation range values

### PR DESCRIPTION
**Pull request by ptitjano**
## Description

Make the widget invalid if the elevation lower value if greater than the upper one.
Capture d’écran du 2024-11-04 13-35-17

### CODE REVIEW

File: src/app/project/qgsprojectelevationsettingswidget.cpp [hunk 1]
```diff
@@ -241,6 +244,19 @@ bool QgsProjectElevationSettingsWidget::validate()
       mMessageBar->pushMessage( tr( "An elevation layer must be selected for a mesh terrain" ), Qgis::MessageLevel::Critical );
     }
   }
+
+  // show an error message if the lower value is greater than the upper one
+  // also show the error message if one of the values is not set
+  // however, do not show the error message if neither is set
+  const bool elevationLowerIsNotSet = mElevationLowerSpin->value() == mElevationLowerSpin->clearValue();
+  const bool elevationUpperIsNotSet = mElevationUpperSpin->value() == mElevationUpperSpin->clearValue();
```
Comment (nyalldawson) [inline src/app/project/qgsprojectelevationsettingswidget.cpp:252]: I think we'd benefit from a QgsDoubleSpinBox::isCleared / isClearValue

**Review by nyalldawson — APPROVED** [review]

File: src/app/project/qgsprojectelevationsettingswidget.cpp [hunk 2]
```diff
@@ -241,6 +244,19 @@ bool QgsProjectElevationSettingsWidget::validate()
       mMessageBar->pushMessage( tr( "An elevation layer must be selected for a mesh terrain" ), Qgis::MessageLevel::Critical );
     }
   }
+
+  // show an error message if the lower value is greater than the upper one
+  // also show the error message if one of the values is not set
```
Comment (uclaros) [inline src/app/project/qgsprojectelevationsettingswidget.cpp:249]:
```
  // also show the error message if one of the values is not set
```
why so?
Can't we have a project with just a minimum or maximum limit?
Even if we couldn't, the error message would be misleading!

File: src/app/project/qgsprojectelevationsettingswidget.cpp (hunk 1 above)
Comment (ptitjano) [reply src/app/project/qgsprojectelevationsettingswidget.cpp]: Done.

File: src/app/project/qgsprojectelevationsettingswidget.cpp (hunk 2 above)
Comment (ptitjano) [reply src/app/project/qgsprojectelevationsettingswidget.cpp]: I have removed the constraint that both limits need to be set.

**Review by lbartoletti — APPROVED** [review]

**Pull request merged**